src/main/scala/it/unibo/authsim/library/user/builder/UserBuilder.scala

```diff
@@ -0,0 +1,34 @@
+package it.unibo.authsim.library.user.builder
+
+import  it.unibo.authsim.library.user.model.User
+import it.unibo.authsim.library.dsl.policy.checkers.StringPolicyChecker
+import it.unibo.authsim.library.dsl.policy.model.StringPolicies.{CredentialPolicy, PasswordPolicy, UserIDPolicy}
+import it.unibo.authsim.library.user.builder.util.Util.generateRandomString
+//ANche qui non sono troppo sicura sull'iso
+abstract class UserBuilder[U]:
+  protected var _credentialPolicies: Seq[CredentialPolicy] = Seq.empty
+  protected var _userName: String=generateRandomString()
+  protected var _password:String=generateRandomString()
+
+  /**
+   * Method that check if the the provided credential for the user meet the input policy
+   * 
+   * @return true if the the credentials are complaint with the policy or false if they are not
+   */
+  protected def checkPolicy(): Boolean=
+    _credentialPolicies.filter(e=> (e.isInstanceOf[PasswordPolicy]|| e.isInstanceOf[UserIDPolicy])).map(e=>
+      if(e.isInstanceOf[PasswordPolicy]) then
+        StringPolicyChecker(e.asInstanceOf[PasswordPolicy]) check _password
+      else
+        StringPolicyChecker(e.asInstanceOf[UserIDPolicy]) check _userName
```

Comment:
```suggestion
      // You may not need "e => e" in the map (you can start directly with "match"), but I'm not sure about this
      e match {
        // I'm not sure about the deletion of the "asInstanceOf", it must be tested
        case PasswordPolicy => StringPolicyChecker(e) check _password
        case UserIDPolicy => StringPolicyChecker(e) check _userName
      }
```

Reply:
I have already tried this, but build fails because it does not whar `PasswordPolicy/UserIDPolicy` are writing something like: `e.isInstanceOf[PasswordPolicy] => StringPolicyChecker(e) check _password` it says something about an invalid pattern if I am not wrong

Reply:
It is not a problem, I was not sure it could be simplified.